operators/pkg/controller/common/user/provided.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package user
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// UserName is the field in the secret that contains the username.
+	UserName = "name"
+	// PasswordHash is the field in the secret that contains the hash of the password.
+	PasswordHash = "passwordHash"
+	// UserRoles contains the roles for the user as a comma separated list of strings.
+	UserRoles = "userRoles"
+
+	fieldNotFound = "field %s not found in secret %s/%s"
+)
+
+// ProvidedUser represents a user that is not created or managed by Elasticsearch.
+// For example in the case of Kibana a user with the right role is provided by the Kibana association controller.
+type ProvidedUser struct {
```

Comment:
I had to think twice to understand what the name `ProvidedUser` means. I am not sure I have a good alternative though. Is `ExternalUser` better? Not sure, maybe others have ideas?